src/main/java/org/tictactoeclass/Main.java

```diff
@@ -19,116 +20,183 @@ public static void main(String[] args) {
+                player1 = new Player(bot.getName(), bot.getSymbol());
+            }
             //Get information about player 2
             System.out.print("Player 2: write your name: ");
             String player2Name = input.nextLine();
             System.out.print("Player 2: Enter your symbol: ");
             char player2Symbol = input.nextLine().charAt(0);
             player2Symbol = SymbolNotDigit(player2Symbol, player2Name);
             //Check the symbol is non-numeric
-            playerSymbol(player1Symbol, player2Symbol, player2Name);
+            playerSymbol(player1.getSymbol(), player2Symbol, player2Name);
             //Instantiate player 2
             player2 = new Player(player2Name, player2Symbol);
 //////////////////////////////////////////////////////////////////////////////
         }
         boolean exit = false;
-        //Only works if the board is not full, no one exit the game and no one win
+
+        // Only works if the board is not full, no one exit the game and no one win
         while (!board.isGridFull() && !board.win() && !exit) {
             String playerName = "Unknown";
-            if (turnCount % 2 == 0) {
-                playerName = player1.getName();
-                board.setSymbol(player1.getSymbol());
-            } else {
-                playerName = player2.getName();
-                board.setSymbol(player2.getSymbol());
-            }
-            System.out.println(playerName + ", Make a move, Enter position (row then column): ");
-            System.out.print("or press 0 0 if you want to exit: ");
-
-            boolean correctInput = false;
             int row = 0;
             int column = 0;
-            while (!correctInput && !exit) {
-                try {
-                    //the user count the row and column from 1, while the program count it from 0
-                    row = Integer.parseInt(input.next()) - 1;
-                    column = Integer.parseInt(input.next()) - 1;
-
-                    //exit game and save objects in file
-                    if (row == -1 && column == -1) {
-                        exit = true;
-                        stateManager.saveState(player1, player2, board, turnCount);
-                    }
-                    //check validity of row and column
-                    while ((row < 0 || row > position.length - 1 || column < 0 || column > position.length - 1) && !exit) {
-                        System.out.println(playerName + ", Invalid position!, numbers from 1-3  ");
-                        System.out.print("or press 0 0 if you want to exit: ");
+            // Bot Action
+            if (numOfPlayer == 1 && turnCount % 2 == 0) {
+                playerName = bot.getName();
+                board.setSymbol(bot.getSymbol());
+                boolean correctRandom=false;
+                while (!correctRandom) {
+                    row = bot.makemove().getRow();
+                    column = bot.makemove().getColumn();
```

Comment:
Maybe calling makemove() here is redundant, since it is already called in the previous line which means you can directly extract it.